server/tests/api_v4/test_enrollment.py

```diff
@@ -17,7 +20,34 @@
         "user",
     ),
 )
-async def test_device_enrollment(invitation: str, coolorg: CoolorgRpcClients):
+@pytest.mark.parametrize(
+    "issue7212",
+    (
+        False,
+        True,
+    ),
+    ids=("normal", "issue7212"),
+)
+async def test_device_enrollment(
+    invitation: str, coolorg: CoolorgRpcClients, issue7212: bool, monkeypatch
+):
+    if issue7212:
+        pg_conduit_listen = PGInviteComponent._conduit_listen
+        memory_conduit_listen = MemoryInviteComponent._conduit_listen
+
+        async def patched_pg_conduit_listen(self, now, ctx, *args, **kwargs):
```

Comment:
Add a comment to explain the goal ?
```
# Make the greeter hang on state 4, so that claimer is the one guaranteed to
# be the first one talking to the conduit.
```

I'm not sure of explanation though. But if it is correct, then shouldn't we divide this test into "claimer first to talk in 4 conduit" and "greeter first to talk conduit" (possibly making the order always the same no matter the state of the conduit, not only for the state 4).

Reply:
[195d06b](https://github.com/Scille/parsec-cloud/pull/7307/commits/195d06bc98e169d4a6f8583ab5f38c1bd1ef9de6)